### DISCUSSION TITLE
Pre-built DuckDB 0.7.0 is not working in OS X Catalina (10.15.7)

### DISCUSSION THREAD

**Discussion by rsund** [duckdb/duckdb#6307] [Q&A] [answered]
I have an old MacBook that won't allow install newer OS X versions than Catalina (10.15.7). Version 0.6.1 of the pre-built DuckDB CLI worked just fine on that machine.

But now with DuckDB version 0.7.0 it throws an error and aborts:

```
dyld: Symbol not found: __ZTTNSt3__114basic_ifstreamIcNS_11char_traitsIcEEEE
  Referenced from: ./duckdb (which was built for Mac OS X 12.6)
  Expected in: /usr/lib/libc++.1.dylib
 in ./duckdb
zsh: abort      ./duckdb
```

Any suggestions for workarounds? Is this something that would be worth of reporting as an issue?

**Comment by Mytherin** [comment, marked as answer]
Looks like an incompatibility in the C++ standard library between the MacOS versions. Perhaps try compiling DuckDB yourself?

**Reply by rsund** [reply]
Thanks! Compiling from source took a while, but was relatively easy task as I had Xcode with command line tools already installed and the only additional step was to install CMake and its command line tools.

Now there is, unfortunately, another nuisance: (out of tree) extensions are not available and I should probably compile also those by myself using the same toolchain.

Any tips for that? Should I try to use build_out_of_tree_extensions.py at the scripts folder or use some environment variables and/or flags to guide make to build those?

**Reply by rsund** [reply]
I had some luck with `build_out_of_tree_extensions.py` after I first manually set up openssl in a same way as in the CI-build for OS X. I also had to change some `nullptr_t` to `std::nullptr_t` manually in the sqlite_scanner -sources, but after that it seemed to work. Still struggling with some extensions, but I need to take a look at those later on.

**Reply by Mause** [reply]
Sounds like we're missing the `mmacosx-version-min` flag? Not sure where exactly we'd need to add that to have it apply to everything though

**Reply by Mause** [reply]
Actually, it might be as simple as setting `CMAKE_OSX_DEPLOYMENT_TARGET` to `10.14` - https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html

**Comment by liquidaty** [comment]
@Mytherin given that the compiled binary is "universal", would a better solution be to statically link the lib or otherwise avoid the dependency, rather than require users to compile themselves? If that is not feasible, it would be helpful to change the name of the binary to specify its minimum requirements. Catalina isn't exactly new, but 3 yrs old isn't all that old either.

**Reply by AlexR2D2** [reply]
I have the same issue in Big Sur 11.7.3. Extensions loading doesn't work :)

"Load fts;" ->

`
IO Error: Extension \"/Users/alex/.duckdb/extensions/v0.7.1/osx_amd64/fts.duckdb_extension\" could not be loaded: dlopen(/Users/alex/.duckdb/extensions/v0.7.1/osx_amd64/fts.duckdb_extension, 6): Symbol not found: __ZNKSt3__115basic_stringbufIcNS_11char_traitsIcEENS_9allocatorIcEEE3strEv\n  Referenced from: /Users/alex/.duckdb/extensions/v0.7.1/osx_amd64/fts.duckdb_extension (which was built for Mac OS X 12.6)\n  Expected in: /usr/lib/libc++.1.dylib\n
`
